electrum/wallet.py

```diff
@@ -1960,22 +1988,39 @@ def fee_estimator(size: Union[int, float, Decimal]) -> int:
             #       forever. see #5433
             # note: Actually, it might be the case that not all UTXOs from the wallet are
             #       being spent if the user manually selected UTXOs.
-            sendable = sum(map(lambda c: c.value_sats(), coins))
-            for (_,i) in i_max:
-                outputs[i].value = 0
+            def distribute_amount(amount):
+                if amount < 0:
+                    raise NotEnoughFunds()
+                distr_amount = 0
+                for (weight, i) in i_max:
+                    # fixme: this does not check that value >= dust_threshold
+                    val = int((amount/i_max_sum) * weight)
+                    outputs[i].value = val
+                    distr_amount += val
+                (x,i) = i_max[-1]
+                outputs[i].value += (amount - distr_amount)
+
+            distribute_amount(0)
             tx = PartialTransaction.from_io(list(coins), list(outputs))
             fee = fee_estimator(tx.estimated_size())
-            amount = sendable - tx.output_value() - fee
-            if amount < 0:
-                raise NotEnoughFunds()
-            distr_amount = 0
-            for (weight, i) in i_max:
-                val = int((amount/i_max_sum) * weight)
-                outputs[i].value = val
-                distr_amount += val
-
-            (x,i) = i_max[-1]
-            outputs[i].value += (amount - distr_amount)
+
+            input_amount = sum(c.value_sats() for c in coins)
+            allocated_amount = sum(o.value for o in outputs if not parse_max_spend(o.value))
+            to_distribute = input_amount - allocated_amount - fee
+            distribute_amount(to_distribute)
+
+            amount_retained = sum(o.value for o in outputs if self.is_mine(o.address))
+            to_be_spent = input_amount - amount_retained # this includes the fee
+
+            if self.should_keep_reserve_utxo(to_be_spent, is_anchor_channel_opening):
+                self.logger.info(f'Adding change output to meet utxo reserve requirements')
```

Comment:
could you add a unit test that enters this branch?
just lie about `is_anchor_channel_opening=True` and mimic one of the existing tests in test_wallet_vertical

Reply:
I added new tests.